app/controllers/reportings_controller.rb

```diff
@@ -200,6 +200,13 @@ def new
   def create
     @reporting = @project.reportings_via_source.build
     @reporting.reporting_to_project_id = params['reporting']['reporting_to_project_id']
+
+    if @reporting.reporting_to_project.nil?
+      flash.now[:error] = l('timelines.reporting_could_not_be_saved')
+      render action: :new, status: :unprocessable_entity
+      return
+    end
+
```

Comment:
can't we rather have a validity check on @reporting before the call to check_visibility and fall back to the error already present in the else case below?

Just for the idea:

```
if @reporting.valid? && check_visibility
  @reporting.save
  ...
else
  flash.now ...
end
``` 

Otherwise we will have duplicated the code of the controller and of the model